### PR TITLE
Inference Pipeline

### PR DESCRIPTION
**Pull request by EricMarcus-ai**
Implements pipeline for inference;
* Working H5 and TiffWriter callbacks
* Inference callback configs
* "New" WSIBatchSampler for Predict -- needed for lightning pipeline

Questions:
* In utils/manifest.py I had to deal with masks/annotations being None (if not filled in the database). There may be a better /neater way of dealing with this. Any ideas?

### CODE REVIEW

File: ahcore/utils/manifest.py [hunk 1]
```diff
@@ -116,12 +116,18 @@ def get_mask_and_annotations_from_record(
...
-    _masks = parse_annotations_from_record(annotations_root, record.masks)
-    _annotations = parse_annotations_from_record(annotations_root, record.annotations)
+    if record.masks:
+        _masks = parse_annotations_from_record(annotations_root, record.masks)
+    else:
+        _masks = None  # should we raise an error here if it is not in predict?
+    if record.annotations:
+        _annotations = parse_annotations_from_record(annotations_root, record.annotations)
+    else:
+        _annotations = None
```
Comment (jonasteuwen) [inline ahcore/utils/manifest.py:126]: I think the parsing is correct, but can't you change parse_annotations_from_record instead?

**Review by jonasteuwen — CHANGES_REQUESTED** [review]
Small changes when checking for annotations.

File: ahcore/utils/manifest.py (hunk 1 above)
Comment (EricMarcus-ai) [reply ahcore/utils/manifest.py]: fixed.

**Review by jonasteuwen — APPROVED** [review]
LGTM

**Pull request merged**